workers/go/kitchensink/kitchen_sink.go

```diff
@@ -39,13 +39,28 @@ func (ca *ClientActivities) ExecuteClientActivity(ctx context.Context, clientAct
 
 type KSWorkflowState struct {
 	workflowState *kitchensink.WorkflowState
+
+	// signal de-duplication fields
+	expectedSignalCount int32
+	expectedSignalIDs   map[int32]interface{}
+	receivedSignalIDs   map[int32]interface{}
 }
 
 func KitchenSinkWorkflow(ctx workflow.Context, params *kitchensink.WorkflowInput) (*common.Payload, error) {
 	workflow.GetLogger(ctx).Debug("Started kitchen sink workflow")
 
 	state := KSWorkflowState{
-		workflowState: &kitchensink.WorkflowState{},
+		workflowState:       &kitchensink.WorkflowState{},
+		expectedSignalCount: 0,
+		expectedSignalIDs:   make(map[int32]interface{}),
+		receivedSignalIDs:   make(map[int32]interface{}),
```

Comment:
Would it be reasonable/possible to only use one map?

Reply:
Done